### DISCUSSION TITLE
Serialize selected config groups at runtime

### DISCUSSION THREAD

**Discussion by addisonklinke** [facebookresearch/hydra#2431] [Q&A] [answered]
Similar to how Hydra automatically tracks `overrides.yaml` - is it possible to determine from an application's code what config groups were selected at runtime? You could attempt to parse the overrides, i.e. ones containing `/` are obviously config groups, but if there's no nesting you can't tell the difference between a group and top-level field

**Comment by Jasha10** [comment, marked as answer]
Yes, see [`hydra.runtime.choices`](https://hydra.cc/docs/1.2/configure_hydra/intro/#hydraruntime).
You can access it using the [hydra resolver](https://hydra.cc/docs/1.2/configure_hydra/intro/#resolvers-provided-by-hydra) (`${hydra:runtime.choices}`) or in your python code as below:

```python
from omegaconf import DictConfig
import hydra
from hydra.core.hydra_config import HydraConfig

@hydra.main(version_base=None, config_path="conf", config_name="config")
def my_app(cfg: DictConfig) -> None:
    print(HydraConfig.get().runtime.choices)

if __name__ == "__main__":
    my_app()
```
